xcube/server/api.py

```diff
@@ -632,10 +659,21 @@ class ApiError(Exception):
     """
 
     def __init__(self,
-                 status_code: int,
+                 status_code: int = 500,
                  message: Optional[str] = None):
         super().__init__(status_code, message)
 
+    BadRequest: Type["_DerivedApiError"]
+    Unauthorized: Type["_DerivedApiError"]
+    Forbidden: Type["_DerivedApiError"]
+    NotFound: Type["_DerivedApiError"]
+    MethodNotAllowed: Type["_DerivedApiError"]
+    Conflict: Type["_DerivedApiError"]
+    Gone: Type["_DerivedApiError"]
+    InternalServerError: Type["_DerivedApiError"]
+    NotImplemented: Type["_DerivedApiError"]
+    InvalidServerConfig: Type["_DerivedApiError"]
```

Comment:
Why are they all of ``Type["_DerivedApiError"]``? This seems wrong to me, because of line 747ff, and ``_BadRequest`` is not a ``_DerivedApiError``.

Reply:
I use the dummy type `_DerivedApiError` because I cannot use `Type["ApiError"]` and this because it has status code as first positional argument and then expressions such as `raise ApiError.BadRequest()` would be shown as error by the IDE's type checker.